src/pie-chart/chart-pie-internal.tsx

```diff
@@ -62,29 +61,24 @@ export const InternalPieChart = forwardRef(
     // We convert pie tooltip options to the core chart's getTooltipContent callback,
     // ensuring no internal types are exposed to the consumer-defined render functions.
     const getTooltipContent: CoreChartProps["getTooltipContent"] = () => {
-      const transformSlotProps = (props: CoreChartProps.TooltipSlotProps): PieChartProps.TooltipDetailsRenderProps => {
-        const point = props.items[0].point;
+      const transformDetailsProps = ({
+        point,
+      }: CoreChartProps.TooltipDetailsProps): PieChartProps.TooltipDetailsRenderProps => {
         return {
           totalValue: point.total ?? 0,
           segmentValue: point.y ?? 0,
           segmentId: getOptionsId(point.options),
           segmentName: point.name ?? "",
         };
       };
+      const transformSlotProps = (props: CoreChartProps.TooltipSlotProps): PieChartProps.TooltipDetailsRenderProps => {
+        const point = props.items[0].point;
+        return transformDetailsProps({ point });
+      };
       return {
         header: tooltip?.header ? (props) => tooltip.header!(transformSlotProps(props)) : undefined,
-        body:
-          tooltip?.body || tooltip?.details
-            ? (props) =>
-                tooltip.body ? (
-                  tooltip.body(transformSlotProps(props))
-                ) : (
-                  <ChartSeriesDetails
-                    details={tooltip?.details?.(transformSlotProps(props)) ?? []}
-                    compactList={true}
-                  />
-                )
-            : undefined,
+        details: tooltip?.details ? (props) => tooltip.details!(transformDetailsProps(props)) : undefined,
```

Comment:
Why are these exclamation marks actually needed if we already checked that the property (`tooltip.details` in this case) is truthy on the left side of the ternary operator?

Reply:
That is because the invocation happens inside a lambda function - so the prev check is ignored by TS